client/components/Modal/PageCreationModal.tsx

```diff
@@ -0,0 +1,174 @@
+import React, { useState, FC } from 'react'
+import { useTranslation } from 'react-i18next'
+import { Button, Form, Input, Label, Modal, ModalBody, ModalHeader } from 'reactstrap'
+import moment from 'moment'
+import styled from 'styled-components'
+
+import { dark, gray, light } from '../../constants/colors'
+import Crowi from 'client/util/Crowi'
+import Icon from '../Common/Icon'
+
+const parentPath = (path: string) => {
+  if (path === '/') {
+    return path
+  }
+
+  if (path.match(/.+\/$/)) {
+    return path
+  }
+
+  return path + '/'
+}
+
+interface Props {
+  crowi: Crowi
+}
+
+const PageCreationModal: FC<Props> = ({ crowi }) => {
+  const { user } = crowi
+  const currentPath = location.pathname
+  const userPath = `/user/${user && user.name}/`
+  const datePath = moment(Date.now()).format('/YYYY/MM/DD/')
+  const isTopPage = currentPath === '/'
+
+  const [t] = useTranslation()
+  const [isOpen, setIsOpen] = useState<boolean>(false)
+  const [portalName, setPotalName] = useState<string>(t('Memo'))
+  const [pageName, setPageName] = useState<string>('')
+  const [underTreePath, setUnderTreePath] = useState<string>(parentPath(currentPath))
+
+  const createTodayPage = async (e: React.FormEvent<HTMLFormElement>) => {
+    e.preventDefault()
+    const href = `${userPath}${portalName || t('Memo')}${pageName ? datePath : datePath.slice(0, -1)}${pageName}`
+    top.location.href = href
+  }
+
+  const createUnderTreePage = async (e: React.FormEvent<HTMLFormElement>) => {
+    e.preventDefault()
+    top.location.href = underTreePath
+  }
+
+  const toggle = () => {
+    setIsOpen(!isOpen)
+  }
+
+  return (
+    <>
+      <NewButton onClick={toggle}>
+        <NewIcon name="pencilOutline" />
+        {t('New')}
+      </NewButton>
+      <Modal isOpen={isOpen} fade={false} toggle={toggle}>
+        <ModalHeader toggle={toggle}>{t('New Page')}</ModalHeader>
+        <ModalBody>
+          <FormLabel>{t("Create today's")}</FormLabel>
+          <Form onSubmit={createTodayPage} inline>
+            <TodayInputBox>
+              <Path>{userPath}</Path>
+              <PortalNameInput type="text" value={portalName} onChange={(event: React.ChangeEvent<HTMLInputElement>) => setPotalName(event.target.value)} />
+              <Path>{datePath}</Path>
+              <PageNameInput
+                type="text"
+                value={pageName}
+                onChange={(event: React.ChangeEvent<HTMLInputElement>) => setPageName(event.target.value)}
+                placeholder={t('Page name (optional)')}
+              />
+            </TodayInputBox>
+            <CreateButton>{t('Create')}</CreateButton>
+          </Form>
+          {!isTopPage && (
+            <>
+              <hr />
+              <FormLabel>{t('Create under', parentPath(currentPath))}</FormLabel>
```

Comment:
Maybe you can refer to this.
https://github.com/crowi/crowi/blob/90e170e4dd7c32f617b975b1fc5a7f002fe108ee/client/components/Admin/App/SecuritySettings.tsx#L88

Reply:
I just fixed now!